### PR TITLE
docs: add JobSet API reference

### PR DESCRIPTION
**Pull request by googs1025**
Fix issue: https://github.com/kubernetes-sigs/jobset/issues/566

How to generate documents:
```bash
root@VM-0-3-ubuntu:/home/ubuntu# cd jobset
root@VM-0-3-ubuntu:/home/ubuntu/jobset# export PATH=$PATH:/usr/local/bin/go/bin
root@VM-0-3-ubuntu:/home/ubuntu/jobset# make generate-apiref
make: go: Permission denied
cd /home/ubuntu/jobset/hack/genref/ && /home/ubuntu/jobset/bin/genref -o /home/ubuntu/jobset/site/content/en/docs/reference
I0701 10:24:47.111766 1227157 main.go:142] Parsing go packages in sigs.k8s.io/jobset/api/jobset/v1alpha2
I0701 10:24:50.923557 1227157 main.go:325] Output written to /home/ubuntu/jobset/site/content/en/docs/reference/jobset.v1alpha2.md
root@VM-0-3-ubuntu:/home/ubuntu/jobset# Connection to 43.132.121.119 closed by remote host.
```

### CODE REVIEW

File: hack/genref/markdown/members.tpl [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{{ define "members" }}
```
Comment (danielvegamyhre) [inline hack/genref/markdown/members.tpl:1]: Are these .tpl files autogenerated?

File: hack/genref/config.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+hiddenMemberFields:
```
Comment (danielvegamyhre) [inline hack/genref/config.yaml:1]: Can you add a comment with a link to the documentation for this config file, so we can easily figure out how to update it later if we need to?

**Comment by googs1025** [comment]
API reference autogenerated  based on this project: https://github.com/kubernetes-sigs/reference-docs/tree/master/genref

File: hack/genref/markdown/members.tpl (hunk 1 above)
Comment (googs1025) [reply hack/genref/markdown/members.tpl:1]: This tql file is added by me, for reference: https://github.com/kubernetes-sigs/kueue/pull/1263

**Comment by danielvegamyhre** [comment]
@googs1025  I checked out this PR and previewed the output docs locally, is the `{#jobset-x-k8s-io-v1alpha2-JobSet}` next to the `JobSet` expected?

image

**Comment by googs1025** [comment]
I checked the API Reference of kueue and got the same result. This is the result of the automatic document generation.
[here](https://github.com/kubernetes-sigs/kueue/blob/main/site/content/en/docs/reference/kueue-alpha.v1alpha1.md)
image

**Comment by danielvegamyhre** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**